ENH: Minor change to parallel_coordinates (#15908)

stangirala opened the pull request:
Add option to sort class lables, add to test

 - [x] closes #15908
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

stangirala committed:
ENH: Minor change to parallel_coordinates (#15908)

Add option to sort class lables, add to test

jreback commented on pandas/tests/plotting/test_misc.py:
```diff
@@ -241,6 +241,22 @@ def test_parallel_coordinates(self):
         with tm.assert_produces_warning(FutureWarning):
             parallel_coordinates(df, 'Name', colors=colors)
 
+        df = DataFrame({"feat": [i for i in range(30)],
```
make this a separate named test

jreback replied:
add the issue number as a comment

stangirala replied:
Done

jreback commented on pandas/tools/plotting.py:
```diff
@@ -718,6 +719,8 @@ def parallel_coordinates(frame, class_column, cols=None, ax=None, color=None,
     axvlines_kwds: keywords, optional
         Options to be passed to axvline method for vertical lines
+    sort_labels: bool, optional
+        Sort class_column labels, useful when assigning colours
```
add a versionadded tag here, add the default (False) instead of optional

jreback commented:
pls add a whatsnew entry, in Other Enhancements.

stangirala committed:
Move feature test to new method, add to whatsnew

Also fix minor bug due to a misnamed var that both the test and pyflakes
missed

TomAugspurger commented:
@stangirala could you rebase and push one more time? Looks like our CI services were having trouble last time you pushed.

stangirala commented:
@TomAugspurger @jreback Looks like the CI builds are passing.

stangirala commented:
@TomAugspurger I'm actually not sure why the code coverage tool is reporting a decrease in coverage.

TomAugspurger commented:
Yeah, I took a glance and couldn't figure it out either. I'm guessing it's just a fluke, but I'll take a closer look later and merge.

stangirala commented:
Thanks!

jreback commented:
if your master is not up to date code coverage doesn't make much sense

jreback commented on doc/source/whatsnew/v0.20.0.txt:
```diff
@@ -371,6 +371,8 @@ Other Enhancements
 - :func:`MultiIndex.remove_unused_levels` has been added to facilitate :ref:`removing unused levels <advanced.shown_levels>`. (:issue:`15694`)
 - ``pd.read_csv()`` will now raise a ``ParserError`` error whenever any parsing error occurs (:issue:`15913`, :issue:`15925`)
 - ``pd.read_csv()`` now supports the ``error_bad_lines`` and ``warn_bad_lines`` arguments for the Python parser (:issue:`15925`)
+- ``pd.read_csv()`` will now raise a ``csv.Error`` error whenever an end-of-file character is encountered in the middle of a data row (:issue:`15913`)
```
this line is duped

stangirala replied:
I don't see this duped on my copy, can you double check?

jreback commented on doc/source/whatsnew/v0.20.0.txt:
```diff
@@ -371,6 +371,8 @@ Other Enhancements
 - ``pd.read_csv()`` will now raise a ``ParserError`` error whenever any parsing error occurs (:issue:`15913`, :issue:`15925`)
 - ``pd.read_csv()`` now supports the ``error_bad_lines`` and ``warn_bad_lines`` arguments for the Python parser (:issue:`15925`)
+- ``pd.read_csv()`` will now raise a ``csv.Error`` error whenever an end-of-file character is encountered in the middle of a data row (:issue:`15913`)
+- ``parallel_coordinates()`` now has a ``sort_labels`` keyword arg that sorts class labels and the colours assigned to them (:issue:`15908`)
```
``has gained a ``

stangirala replied:
Fixed.

jreback commented on pandas/tests/plotting/test_misc.py:
```diff
@@ -241,6 +241,26 @@ def test_parallel_coordinates(self):
 
+    def test_parallel_coordinates_with_sorted_labels(self):
+        """ For #15908 """
+        from pandas.tools.plotting import parallel_coordinates
```
you may need a mpl skip here? @TomAugspurger ?

TomAugspurger replied:
Done on the class.

jreback commented on pandas/tools/plotting.py:
```diff
@@ -774,6 +780,9 @@ def parallel_coordinates(frame, class_column, cols=None, ax=None, color=None,
                                         colormap=colormap, color_type='random',
                                         color=color)
 
+    if sort_labels is True:
```
``if sort_labels`` is enough

stangirala replied:
I mirrored an if expression from the lines above, but fixed.

jreback commented:
lgtm. minor comment.

stangirala committed:
Add minor code change, what's new doc fix

jreback commented:
thanks @stangirala
